api/policy_exceptions.go

```diff
@@ -0,0 +1,109 @@
+// the Exceptions schema from the Lacework APIv2 Server
+type policyExceptionsService struct {
+	client *Client
+}
+
+// List returns a list of the Policy Exceptions for a policy ID.
+func (svc policyExceptionsService) List(policyID string) (response PolicyExceptionsResponse, err error) {
+	if policyID == "" {
+		return response, errors.New("specify a policy ID")
+	}
+	err = svc.client.RequestDecoder("GET", fmt.Sprintf(apiV2PolicyExceptions, policyID), nil, &response)
+	return
+}
+
+// Get returns a raw response of the Policy Exception with the matching policy ID and exception ID.
+func (svc policyExceptionsService) Get(policyID string, exceptionID string, response interface{}) error {
+	if exceptionID == "" || policyID == "" {
+		return errors.New("specify exception and policy IDs")
+	}
+	apiPath := fmt.Sprintf(apiV2PolicyExceptionsFromExceptionID, exceptionID, policyID)
+	return svc.client.RequestDecoder("GET", apiPath, nil, &response)
+}
+
+func (svc policyExceptionsService) Delete(policyID string, exceptionID string) error {
+	if exceptionID == "" || policyID == "" {
+		return errors.New("specify exception and policy IDs")
+	}
+
+	return svc.client.RequestDecoder(
+		"DELETE",
+		fmt.Sprintf(apiV2PolicyExceptionsFromExceptionID, exceptionID, policyID),
+		nil,
+		nil,
+	)
+}
+
+// Create creates a single Policy Exception
+func (svc *policyExceptionsService) Create(policyID string, policy PolicyException) (
+	response PolicyExceptionResponse,
+	err error,
+) {
+	if policyID == "" {
+		return response, errors.New("specify a policy ID")
+	}
+	err = svc.client.RequestEncoderDecoder("POST", fmt.Sprintf(apiV2PolicyExceptions, policyID),
+		policy, &response)
+	return
+}
+
+// Update updates a single Policy Exception
+func (svc policyExceptionsService) Update(policyID string, exception PolicyException) (response PolicyExceptionResponse, err error) {
+	if exception.ExceptionID == "" || policyID == "" {
+		return response, errors.New("specify exception and policy IDs")
+	}
+	apiPath := fmt.Sprintf(apiV2PolicyExceptionsFromExceptionID, exception.ExceptionID, policyID)
+	// Request is invalid if it contains the ExceptionID, LastUpdatedTime or LastUpdatedUser fields.
+	exception.ExceptionID = ""
+	exception.LastUpdateUser = ""
+	exception.LastUpdateTime = ""
+
+	err = svc.client.RequestEncoderDecoder("PATCH", apiPath, exception, &response)
+	return
+}
+
+type PolicyExceptionResponse struct {
+	Data PolicyException `json:"data"`
+}
+
+type PolicyExceptionsResponse struct {
+	Data []PolicyException `json:"data"`
+}
+type PolicyException struct {
+	ExceptionID    string                      `json:"exceptionId,omitempty"`
+	Description    string                      `json:"description"`
+	Constraints    []PolicyExceptionConstraint `json:"constraints"`
+	LastUpdateTime string                      `json:"lastUpdateTime,omitempty"`
+	LastUpdateUser string                      `json:"lastUpdateUser,omitempty"`
```

Comment:
can these fields be `time.Time`?

Reply:
LastUpdateTime, probably yeh. Though , from looking at our other lastupdatedtime fields we often use string. LastUpdateTime is only returned in the response, in terraform it'll be a computed value. A user will never have to construct a date for this field in a request. If we want to have this field also as a date type, I think having a method on PolicyException struct might be a good idea instead